master/docs/developer/utils.rst

```diff
@@ -1040,3 +1040,110 @@ For example, a particular daily scheduler could be configured on multiple master
+        :param base_url: The base http url of the service to access. e.g. ``http://github.com/``
+        :param auth: Authentication information. If auth is a tuple then ``BasicAuth`` will be used. e.g ``('user', 'passwd')``
+            It can also be a :mod:`requests.auth` authentication plugin.
+            In this case `txrequests`_ will be forced, and `treq`_ cannot be used.
+        :returns: instance of :`HTTPClientService`
+
+        Get an instance of the SharedService.
+        There is one instance per base_url and auth.
+
+        The constructor initialize the service, and store the config arguments in private attributes.
+
+        This should *not* be overriden by subclasses, as they should rather override checkConfig.
+
+
+    .. py:method:: get(endpoint, params=None)
+
+        :param endpoint: endpoint relative to the base_url (starts with ``/``)
+        :param params: optional dictionary that will be encoded in the query part of the url (e.g. ``?param1=foo``)
+        :returns: implementation of :`IHTTPResponse` via deferred
+
+        Performs a HTTP ``GET``
+
+    .. py:method:: delete(endpoint, params=None)
+
+        :param endpoint: endpoint relative to the base_url (starts with ``/``)
+        :param params: optional dictionary that will be encoded in the query part of the url (e.g. ``?param1=foo``)
+        :returns: implementation of :`IHTTPResponse` via deferred
+
+        Performs a HTTP ``DELETE``
+
+    .. py:method:: post(endpoint, data=None, json=None, params=None)
+
+        :param endpoint: endpoint relative to the base_url (starts with ``/``)
+        :param data: optional dictionary that will be encoded in the body of the http requests as ``application/x-www-form-urlencoded``
+        :param json: optional dictionary that will be encoded in the body of the http requests as ``application/json``
+        :param params: optional dictionary that will be encoded in the query part of the url (e.g. ``?param1=foo``)
+        :returns: implementation of :`IHTTPResponse` via deferred
+
+        Performs a HTTP ``POST``
+
+        .. note::
+
+            json and data cannot be used at the same time.
+
+    .. py:method:: put(endpoint, data=None, json=None, params=None)
+
+        :param endpoint: endpoint relative to the base_url (starts with ``/``)
+        :param data: optional dictionary that will be encoded in the body of the http requests as ``application/x-www-form-urlencoded``
+        :param json: optional dictionary that will be encoded in the body of the http requests as ``application/json``
+        :param params: optional dictionary that will be encoded in the query part of the url (e.g. ``?param1=foo``)
+        :returns: implementation of :`IHTTPResponse` via deferred
+
+        Performs a HTTP ``PUT``
+
+        .. note::
+
+            json and data cannot be used at the same time.
+
+.. py:class:: IHTTPResponse
```

Comment:
It is already mentioned:

        The api is preferably a subset of the `treq`_ API, so that `treq`_ can be implemented as passthrough.

I'll try to rephrase to make it clearer